### DISCUSSION TITLE
Add feature to use `regex-lite` instead?

### DISCUSSION THREAD

**Discussion by yujincheng08** [actix/actix-web#3290] [Q&A] [answered]
Hi, I am using `actix-web` for my project and it works like a charm. However, I have a tiny concern about the binary size. I used `cargo bloat` and figured out that `regex` contributes a large part to the binary size, which is used by `actix-web`. I found out that `rust-lang/regex` offers an alternative to `regex` with smaller binary size but with slower performance and limited Unicode support. I wonder if it's possible to add a feature to use `regex-lite` for users with binary size concerns? I tried to migrate to `regex-lite` and it's quite easy. If it's possible, I am more than happy to open a PR.

**Comment by robjtede** [comment, marked as answer]
I'd be happy to see a PR for this change, though we should be careful and consider Rust best practices when it comes to feature flags being additive.

Edit: PR from @yujincheng08 https://github.com/actix/actix-web/pull/3291
